### PR TITLE
fix: preload=metadata with autoplay

### PR DESCRIPTION
**Pull request by gkatsev**
With autoplay, it is possible to get the play event before
MANIFEST_LOADED happens. In those cases, we don't want to call
`hls.stopLoad` as we got a signal to continue playback.

### CODE REVIEW

File: packages/playback-core/src/index.ts [hunk 1]
```diff
@@ -373,8 +373,14 @@ export const loadMedia = (
...
-        // when preload is metadata, stop loading when the manifest is loaded
-        hls.once(Hls.Events.MANIFEST_LOADED, () => hls.stopLoad());
+        // when preload is metadata, stop loading when the manifest is parsed,
+        hls.once(Hls.Events.MANIFEST_LOADED, () => {
+          // unless we're no longer paused, which can happen with autoplay,
+          // where play will happen before MANIFEST_LOADED ends up triggering
+          if (mediaEl.paused) {
+            hls.stopLoad();
+          }
+        });
```
Comment (luwes) [inline packages/playback-core/src/index.ts:383]: 👍

File: packages/playback-core/src/autoplay.ts [hunk 2]
```diff
@@ -91,18 +91,34 @@ export const setupAutoplay = (
...
+        if (mediaEl.preload === 'metadata') {
+          if (hls) {
+            hls.once(Hls.Events.LEVEL_UPDATED, handleSeek);
+          } else if (mediaEl.seekable.end(0) === Infinity) {
+            const interval = setInterval(() => {
+              if (mediaEl.seekable.end(0) !== Infinity) {
+                clearInterval(interval);
+                handleSeek();
+              }
+            }, 100);
```
Comment (luwes) [inline packages/playback-core/src/autoplay.ts:118]: could we use an event here instead of a setInterval?

Comment (gkatsev) [reply packages/playback-core/src/autoplay.ts:118]: Let me see if there's something reliable here, as that would definitely be better.

**Comment by gkatsev** [comment]
Follow-up:
- fix the test so that we don't skip it anymore
- look into an event that can be used instead of the set interval

**Review by cjpillsbury — APPROVED** [review]
👍

**Pull request merged**